src/main/java/org/dependencytrack/resources/v1/BadgeResource.java

```diff
@@ -114,7 +114,73 @@ public Response getProjectVulnerabilitiesBadge(
                 if (project != null) {
                     final ProjectMetrics metrics = qm.getMostRecentProjectMetrics(project);
                     final Badger badger = new Badger();
-                    return Response.ok(badger.generate(metrics)).build();
+                    return Response.ok(badger.generateVulnerabilities(metrics)).build();
+                } else {
+                    return Response.status(Response.Status.NOT_FOUND).entity("The project could not be found.").build();
+                }
+            } else {
+                return Response.status(Response.Status.NO_CONTENT).build();
+            }
+        }
+    }
+
+    @GET
+    @Path("/violations/project/{uuid}")
+    @Produces(SVG_MEDIA_TYPE)
+    @ApiOperation(
+            value = "Returns current metrics for a specific project",
+            response = ProjectMetrics.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "Badge support is disabled. No content will be returned."),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 404, message = "The project could not be found")
+    })
+    @AuthenticationNotRequired
+    public Response getProjectPolicyViolationsBadge(
+            @ApiParam(value = "The UUID of the project to retrieve metrics for", required = true)
+            @PathParam("uuid") String uuid) {
+        try (QueryManager qm = new QueryManager()) {
+            if (isBadgeSupportEnabled(qm)) {
+                final Project project = qm.getObjectByUuid(Project.class, uuid);
+                if (project != null) {
+                    final ProjectMetrics metrics = qm.getMostRecentProjectMetrics(project);
+                    final Badger badger = new Badger();
+                    return Response.ok(badger.generateViolations(metrics)).build();
+                } else {
+                    return Response.status(Response.Status.NOT_FOUND).entity("The project could not be found.").build();
+                }
+            } else {
+                return Response.status(Response.Status.NO_CONTENT).build();
+            }
+        }
+    }
+
+    @GET
+    @Path("/violations/project/{name}/{version}")
+    @Produces(SVG_MEDIA_TYPE)
+    @ApiOperation(
+            value = "Returns current metrics for a specific project",
+            response = ProjectMetrics.class
```

Comment:
Same as above, please change to a description that fits to what this endpoint actually does. Also change `response` to `String.class` or omit `response` entirely.

Reply:
Also corrected.